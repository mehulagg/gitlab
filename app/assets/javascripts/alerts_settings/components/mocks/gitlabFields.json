[
  {
    "name": "TITLE",
    "label": "Title",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ],
    "numberOfFallbacks": 1
  },
  {
    "name": "DESCRIPTION",
    "label": "Description",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "START_TIME",
    "label": "Start time",
    "type": [
      "DATETIME"
    ],
    "compatibleTypes": [
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "END_TIME",
    "label": "End time",
    "type": [
      "DATETIME"
    ],
    "compatibleTypes": [
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "SERVICE",
    "label": "Service",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "MONITORING_TOOL",
    "label": "Monitoring tool",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "HOSTS",
    "label": "Hosts",
    "type": [
      "STRING",
      "ARRAY"
    ],
    "compatibleTypes": [
      "STRING",
      "ARRAY",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "SEVERITY",
    "label": "Severity",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "FINGERPRINT",
    "label": "Fingerprint",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  },
  {
    "name": "GITLAB_ENVIRONMENT_NAME",
    "label": "Environment",
    "type": [
      "STRING"
    ],
    "compatibleTypes": [
      "STRING",
      "NUMBER",
      "DATETIME"
    ]
  }
]
