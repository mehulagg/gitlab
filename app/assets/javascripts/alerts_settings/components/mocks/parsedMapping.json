[
  {
    "name":"title",
    "label":"Title",
    "type":"String"
  },
  {
    "name":"description",
    "label":"Description",
    "type":"String"
  },
  {
    "name":"startTime",
    "label":"Start time",
    "type":"DateTime"
  },
  {
    "name":"service",
    "label":"Service",
    "type":"String"
  },
  {
    "name":"monitoringTool",
    "label":"Monitoring tool",
    "type":"String"
  },
  {
    "name":"hosts",
    "label":"Hosts",
    "type":"String or Array"
  },
  {
    "name":"severity",
    "label":"Severity",
    "type":"String"
  },
  {
    "name":"fingerprint",
    "label":"Fingerprint",
    "type":"String"
  },
  {
    "name":"environment",
    "label":"Environment",
    "type":"String"
  }
]
