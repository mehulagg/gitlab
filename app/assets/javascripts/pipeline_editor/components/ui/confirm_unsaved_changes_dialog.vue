<script>
export default {
  props: {
    hasUnsavedChanges: {
      type: Boolean,
      required: true,
    },
  },
  created() {
    window.addEventListener('beforeunload', this.confirmChanges);
  },
  destroyed() {
    window.removeEventListener('beforeunload', this.confirmChanges);
  },
  methods: {
    confirmChanges(e) {
      if (this.hasUnsavedChanges) {
        e.preventDefault();
        e.returnValue = '';
      }
    },
  },
  render: () => null,
};
</script>
