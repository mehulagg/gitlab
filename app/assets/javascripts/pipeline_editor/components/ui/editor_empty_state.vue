<script>
import { GlSprintf } from '@gitlab/ui';
import { s__ } from '~/locale';

export default {
  components: {
    GlSprintf,
  },
  i18n: {
    title: s__('PipelineEditor|Optimize your workflow with CI/CD Pipelines'),
    body: s__(
      'PipelineEditor|Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
  },
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
  },
};
</script>
<template>
  <div class="gl-display-flex gl-flex-direction-column gl-align-items-center">
    <img :src="emptyStateIllustrationPath" data-testid="empty-state-illustration" />
    <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
    <p>
      <gl-sprintf :message="$options.i18n.body" class="gl-display-inline">
        <template #code="{ content }">
          <code>{{ content }}</code>
        </template>
      </gl-sprintf>
    </p>
  </div>
</template>
