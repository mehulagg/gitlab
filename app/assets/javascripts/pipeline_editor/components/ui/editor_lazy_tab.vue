<script>
import { GlTab } from '@gitlab/ui';

/**
 * Wrapper of <gl-tab> to lazily render this tab's content
 * when shown **without dismounting it after**.
 *
 * Once the tab is selected it is permanently set as "not-lazy"
 * so it's contents are not dismounted.
 *
 * Usage:
 *
 * Same as <gl-tab>, for example:
 *
 * <gl-tabs>
 *   <editor-lazy-tab title="Tab 1"> ... </editor-lazy-tab>
 *   <editor-lazy-tab title="Tab 2"> ... </editor-lazy-tab>
 * </gl-tabs>
 */
export default {
  components: {
    GlTab,
  },
  props: {
    lazy: {
      type: Boolean,
      required: false,
      default: true,
    },
  },
  data() {
    return {
      isLazy: this.lazy,
    };
  },
  methods: {
    onClick() {
      this.isLazy = false;
    },
  },
};
</script>
<template>
  <gl-tab :lazy="isLazy" v-bind="$attrs" v-on="$listeners" @click="onClick">
    <slot v-for="slot in Object.keys($slots)" :slot="slot" :name="slot"></slot>
  </gl-tab>
</template>
