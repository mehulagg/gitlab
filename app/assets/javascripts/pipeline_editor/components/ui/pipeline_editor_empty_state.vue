<script>
import { GlButton, GlSprintf } from '@gitlab/ui';
import { __ } from '~/locale';
import glFeatureFlagsMixin from '~/vue_shared/mixins/gl_feature_flags_mixin';

export default {
  components: {
    GlButton,
    GlSprintf,
  },
  i18n: {
    title: __('Optimize your workflow with CI/CD Pipelines'),
    body: __(
      'Create a new %{codeStart}.gitlab-ci.yml%{codeEnd} file at the root of the repository to get started.',
    ),
    btnText: __('Create new CI/CD pipeline'),
  },
  mixins: [glFeatureFlagsMixin()],
  inject: {
    emptyStateIllustrationPath: {
      default: '',
    },
  },
  computed: {
    showCTAButton() {
      return this.glFeatures.pipelineEditorEmptyStateAction;
    },
  },
  methods: {
    createEmptyConfigFile() {
      this.$emit('createEmptyConfigFile');
    },
  },
};
</script>
<template>
  <div class="gl-display-flex gl-flex-direction-column gl-align-items-center gl-mt-11">
    <img :src="emptyStateIllustrationPath" />
    <h1 class="gl-font-size-h1">{{ $options.i18n.title }}</h1>
    <p class="gl-mt-3">
      <gl-sprintf :message="$options.i18n.body">
        <template #code="{ content }">
          <code>{{ content }}</code>
        </template>
      </gl-sprintf>
    </p>
    <gl-button
      v-if="showCTAButton"
      variant="confirm"
      class="gl-mt-3"
      @click="createEmptyConfigFile"
    >
      {{ $options.i18n.btnText }}
    </gl-button>
  </div>
</template>
