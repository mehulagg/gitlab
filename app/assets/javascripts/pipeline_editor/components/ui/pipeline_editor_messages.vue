<script>
import { GlAlert } from '@gitlab/ui';
import { getParameterValues, removeParams } from '~/lib/utils/url_utility';
import { __, s__ } from '~/locale';
import {
  COMMIT_FAILURE,
  COMMIT_SUCCESS,
  DEFAULT_FAILURE,
  LOAD_FAILURE_UNKNOWN,
} from '../../constants';
import CodeSnippetAlert from '../code_snippet_alert/code_snippet_alert.vue';
import {
  CODE_SNIPPET_SOURCE_URL_PARAM,
  CODE_SNIPPET_SOURCES,
} from '../code_snippet_alert/constants';

export default {
  components: {
    GlAlert,
    CodeSnippetAlert,
  },
  errorTexts: {
    [COMMIT_FAILURE]: s__('Pipelines|The GitLab CI configuration could not be updated.'),
    [DEFAULT_FAILURE]: __('Something went wrong on our end.'),
    [LOAD_FAILURE_UNKNOWN]: s__('Pipelines|The CI configuration was not loaded, please try again.'),
  },
  successTexts: {
    [COMMIT_SUCCESS]: __('Your changes have been successfully committed.'),
  },
  props: {
    failureType: {
      type: String,
      required: false,
      default: null,
    },
    failureReasons: {
      type: Array,
      required: false,
      default: () => [],
    },
    successType: {
      type: String,
      required: false,
      default: null,
    },
  },
  data() {
    return {
      codeSnippetCopiedFrom: '',
      showAlert: false,
    };
  },
  computed: {
    alerts() {
      const messages = [];

      if (this.failureType) {
        messages.push(this.failure);
      }

      if (this.success) {
        messages.push(this.success);
      }
      return messages;
    },
    failure() {
      switch (this.failureType) {
        case LOAD_FAILURE_UNKNOWN:
          return {
            text: this.$options.errorTexts[LOAD_FAILURE_UNKNOWN],
            variant: 'danger',
          };
        case COMMIT_FAILURE:
          return {
            text: this.$options.errorTexts[COMMIT_FAILURE],
            variant: 'danger',
          };
        default:
          return {
            text: this.$options.errorTexts[DEFAULT_FAILURE],
            variant: 'danger',
          };
      }
    },
  },
  success() {
    switch (this.successType) {
      case COMMIT_SUCCESS:
        return {
          text: this.$options.successTexts[COMMIT_SUCCESS],
          variant: 'info',
        };
      default:
        return null;
    }
  },

  created() {
    this.parseCodeSnippetSourceParam();
  },
  methods: {
    dismissAlert() {
      this.showAlert = false;
    },
    dismissCodeSnippetAlert() {
      this.codeSnippetCopiedFrom = '';
    },
    parseCodeSnippetSourceParam() {
      const [codeSnippetCopiedFrom] = getParameterValues(CODE_SNIPPET_SOURCE_URL_PARAM);
      if (codeSnippetCopiedFrom && CODE_SNIPPET_SOURCES.includes(codeSnippetCopiedFrom)) {
        this.codeSnippetCopiedFrom = codeSnippetCopiedFrom;
        window.history.replaceState(
          {},
          document.title,
          removeParams([CODE_SNIPPET_SOURCE_URL_PARAM]),
        );
      }
    },
  },
};
</script>
<template>
  <div>
    <code-snippet-alert
      v-if="codeSnippetCopiedFrom"
      :source="codeSnippetCopiedFrom"
      class="gl-mb-5"
      @dismiss="dismissCodeSnippetAlert"
    />
    <gl-alert
      v-for="alert in alerts"
      :key="alert.text"
      :variant="alert.variant"
      class="gl-mb-5"
      @dismiss="dismissAlert"
    >
      {{ alert.text }}
      <ul v-if="failureReasons.length" class="gl-mb-0">
        <li v-for="reason in failureReasons" :key="reason">{{ reason }}</li>
      </ul>
    </gl-alert>
  </div>
</template>
