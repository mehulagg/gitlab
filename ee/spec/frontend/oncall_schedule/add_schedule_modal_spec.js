import { shallowMount } from '@vue/test-utils';
import { GlModal, GlAlert } from '@gitlab/ui';
import waitForPromises from 'helpers/wait_for_promises';
import AddScheduleModal from 'ee/oncall_schedules/components/add_schedule_modal.vue';
import { getOncallSchedulesQueryResponse } from './mocks/apollo_mock';
import mockTimezones from './mocks/mockTimezones.json';

describe('AddScheduleModal', () => {
  let wrapper;
  const projectPath = 'group/project';
  const mutate = jest.fn();
  const mockHideModal = jest.fn();

  const createComponent = ({ data = {}, props = {} } = {}) => {
    wrapper = shallowMount(AddScheduleModal, {
      data() {
        return {
          form:
            getOncallSchedulesQueryResponse.data.project.incidentManagementOncallSchedules.nodes[0],
          ...data,
        };
      },
      propsData: {
        modalId: 'modalId',
        ...props,
      },
      provide: {
        projectPath,
        timezones: mockTimezones,
      },
      mocks: {
        $apollo: {
          mutate,
        },
      },
    });

    wrapper.vm.$refs.createScheduleModal.hide = mockHideModal;
  };

  beforeEach(() => {
    createComponent();
  });

  afterEach(() => {
    wrapper.destroy();
    wrapper = null;
  });

  const findModal = () => wrapper.find(GlModal);
  const findAlert = () => wrapper.find(GlAlert);

  it('renders modal layout', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('Schedule create', () => {
    it('makes a request with form data to create a schedule', () => {
      mutate.mockResolvedValueOnce({});
      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      expect(mutate).toHaveBeenCalledWith({
        mutation: expect.any(Object),
        variables: { oncallScheduleCreateInput: expect.objectContaining({ projectPath }) },
      });
    });

    it('hides the modal on successful schedule creation', async () => {
      mutate.mockResolvedValueOnce({ data: { oncallScheduleCreate: { errors: [] } } });
      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      await waitForPromises();
      expect(mockHideModal).toHaveBeenCalled();
    });

    it("doesn't hide a modal and shows error alert on fail", async () => {
      const error = 'some error';
      mutate.mockResolvedValueOnce({ data: { oncallScheduleCreate: { errors: [error] } } });
      findModal().vm.$emit('primary', { preventDefault: jest.fn() });
      await waitForPromises();
      const alert = findAlert();
      expect(mockHideModal).not.toHaveBeenCalled();
      expect(alert.exists()).toBe(true);
      expect(alert.text()).toContain(error);
    });
  });
});
