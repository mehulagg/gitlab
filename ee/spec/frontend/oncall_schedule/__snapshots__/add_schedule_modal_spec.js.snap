// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddScheduleModal renders modal layout 1`] = `
<gl-modal-stub
  actioncancel="[object Object]"
  actionprimary="[object Object]"
  modalclass=""
  modalid="modalId"
  size="sm"
  title="Add schedule"
  titletag="h4"
>
  <!---->
   
  <add-edit-schedule-form-stub
    form="[object Object]"
    schedule="[object Object]"
  />
</gl-modal-stub>
`;
